doc/user/client.md

```diff
@@ -0,0 +1,252 @@
+# Operator-SDK: Controller Runtime Client API
+
+## Overview
+
+The [kubernetes-sigs][org-kubernetes-sigs] organization builds tools to call the [Kubernetes][site-kubernetes] API in a clean, abstracted manner, much like the Operator SDK. The [controller-runtime][repo-controller-runtime] project is meant to help users build [controllers][site-kubernetes-controllers] easily by generating a controller project that interacts with a k8s cluster via CRUD operations. User-defined controller can perform specific tasks in a cluster; [operators][site-operators] can use multiple controllers to perform a variety of tasks. As operators use at least one controller, the SDK can rely on controller-runtime's k8s API code rather than develop a parallel set of API calls to execute the same cluster operations, namely CRUD operations.
```

Comment:
How about making this section of the overview a bit more concise and specific to the controller-runtime's Client. 

Right now this is giving a broad overview of several things plus it's going more into the motivation of why the SDK uses the controller-runtime. That's explanation is more suited for a proposal.

How about something like:
```
The controller-runtime library provides various abstractions to build controllers to watch and reconcile resources.
One of these abstractions is a generic client interface that let's a controller perform CRUD operations for different resource types.
```